### PR TITLE
Fix enabling only `dns-over-*` feature

### LINKED ISSUE #1832: [BUG] Build failure with only feature dns-over-https
`cargo check --no-default-features --features dns-over-https` failed with:

```
warning: /Users/nobodyxu/Dev/trust-dns/bin/Cargo.toml: file found to be present in multiple build targets: /Users/nobodyxu/Dev/trust-dns/bin/src/named.rs
    Checking trust-dns-proto v0.22.0 (/Users/nobodyxu/Dev/trust-dns/crates/proto)
error[E0433]: failed to resolve: use of undeclared crate or module `tokio_rustls`
  --> crates/proto/src/https/https_client_stream.rs:26:5
   |
26 | use tokio_rustls::{
   |     ^^^^^^^^^^^^ use of undeclared crate or module `tokio_rustls`

error[E0432]: unresolved import `rustls`
  --> crates/proto/src/https/https_client_stream.rs:25:5
   |
25 | use rustls::ClientConfig;
   |     ^^^^^^ use of undeclared crate or module `rustls`

error[E0432]: unresolved import `tokio_rustls`
  --> crates/proto/src/https/https_client_stream.rs:26:5
   |
26 | use tokio_rustls::{
   |     ^^^^^^^^^^^^ use of undeclared crate or module `tokio_rustls`

error[E0432]: unresolved import `crate::iocompat`
  --> crates/proto/src/https/https_client_stream.rs:32:12
   |
32 | use crate::iocompat::AsyncIoStdAsTokio;
   |            ^^^^^^^^ could not find `iocompat` in the crate root

error[E0433]: failed to resolve: use of undeclared crate or module `tokio`
   --> crates/proto/src/https/https_client_stream.rs:477:21
    |
477 |                     tokio::spawn(
    |                     ^^^^^ use of undeclared crate or module `tokio`

warning: unused import: `TryFutureExt`
  --> crates/proto/src/https/https_client_stream.rs:20:39
   |
20 | use futures_util::future::{FutureExt, TryFutureExt};
   |                                       ^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

Some errors have detailed explanations: E0432, E0433.
For more information about an error, try `rustc --explain E0432`.
warning: `trust-dns-proto` (lib) generated 1 warning
error: could not compile `trust-dns-proto` due to 5 previous errors; 1 warning emitted
```

I found this when running `cargo hack check --feature-powerset`.

**System:**
This bug does not depend on specific system configure to reproduce.

**Version:**
Crate: proto
Version: 0.22.0

**Additional context**

This bug is originally found in https://github.com/cargo-bins/cargo-binstall/pull/538 and https://github.com/cargo-bins/cargo-binstall/pull/430
It can be reproduced on GHA mac (x86-64)/windows/ubuntu whenever trust-dns is enabled and also on my M1 mac.

### PR DESCRIPTION
**Pull request by NobodyXu**
Fixed #1832

Signed-off-by: Jiahao XU <Jiahao_XU@outlook.com>

### CODE REVIEW

**Comment by NobodyXu** [comment]
P.S. I think it's a good idea to run `cargo hack check --feature-powerset` on GHA to detect this kind of regression.

File: crates/proto/Cargo.toml [hunk 1]
```diff
@@ -42,8 +42,8 @@ dns-over-rustls = ["dns-over-tls", "rustls", "rustls-pemfile", "tokio-rustls", "
 dns-over-native-tls = ["dns-over-tls", "native-tls", "tokio-native-tls"]
 dns-over-openssl = ["dns-over-tls", "openssl", "tokio-openssl"]
 
-dns-over-https-rustls = ["dns-over-https", "dns-over-rustls", "webpki-roots"]
-dns-over-https = ["bytes", "dns-over-tls", "h2", "http"]
+dns-over-https-rustls = ["dns-over-https"]
+dns-over-https = ["bytes", "h2", "http", "dns-over-rustls", "webpki-roots", "tokio-runtime"]
```
Comment (djc) [inline crates/proto/Cargo.toml:46]: This doesn't look right to me, I think `dns-over-https` should only have TLS-implementation-neutral dependencies.

Comment (NobodyXu) [reply crates/proto/Cargo.toml:46]: Well, in practice, the https implementation in trust-dns-resolver depends on rustls and is tightly coupled with it.
If `dns-over-https` is the only feature enabled, then the compiler would give error on missing imports from `rustls`.

Comment (djc) [reply crates/proto/Cargo.toml:46]: Hmm, okay, in that case this probably makes sense.

**Review by djc — APPROVED** [review]

**Pull request merged**